TypeCobol/Compiler/Scanner/ScannerStep.cs

```diff
@@ -16,101 +15,74 @@ public static class ScannerStep
-            [CanBeNull] MultilineScanState initialScanState)
+        public static void ScanDocument(TextSourceInfo textSourceInfo, ISearchableReadOnlyList<TokensLine> documentLines, TypeCobolOptions compilerOptions, List<RemarksDirective.TextNameVariation> copyTextNameVariations, [NotNull] MultilineScanState initialScanState)
         {
             TokensLine tokensLine = null;
             MultilineScanState lastScanState = initialScanState;
             IList<TokensLine> lineGroup = new List<TokensLine>();
             bool groupIsContinuationGroup = false;
 
-            // Get the first line
-            IEnumerator<TokensLine> documentLinesEnumerator = documentLines.GetEnumerator();
-            if (documentLinesEnumerator.MoveNext())
+            using (IEnumerator<TokensLine> documentLinesEnumerator = documentLines.GetEnumerator())
             {
-                tokensLine = documentLinesEnumerator.Current;
-            }
-
-            while (tokensLine != null)
-            {
-                lineGroup.Add(tokensLine);
-
-                //Peek at next line type to decide whether to scan the current line group or continue to accumulate
-                var nextTokensLine = documentLinesEnumerator.MoveNext() ? documentLinesEnumerator.Current : null;
-                switch (nextTokensLine?.Type)
+                // Get the first line
+                if (documentLinesEnumerator.MoveNext())
                 {
-                    case null:
-                    case CobolTextLineType.Source:
-                        //tokensLine is the last one or next line is a new Source line
-                        //Scan the current group
-                        ScanGroup();
-                        break;
-                    case CobolTextLineType.Continuation:
-                        //Remember that the current group contains at least one continuation line
-                        groupIsContinuationGroup = true;
-                        break;
-                    //default: keep accumulating lines (Blank, Comment, etc) into current group
+                    tokensLine = documentLinesEnumerator.Current;
                 }
 
-                tokensLine = nextTokensLine;
-
-                void ScanGroup()
+                while (tokensLine != null)
                 {
-                    if (groupIsContinuationGroup)
+                    lineGroup.Add(tokensLine);
+
+                    //Peek at next line type to decide whether to scan the current line group or continue to accumulate
+                    var nextTokensLine = documentLinesEnumerator.MoveNext() ? documentLinesEnumerator.Current : null;
+                    switch (nextTokensLine?.Type)
                     {
-                        //Group must be scanned as a whole
-                        if (lastScanState == null)
```

Comment:
`if (lastScanState == null)` has been removed
`lastScanState` cannot be `null` or is `null` handled by called method ?

Reply:
`lastScanState` is expected to be non-null all along the process. It is initialized with non-null value passed by the caller (and parameter has the `NotNull` attribute) then values are taken from lines so they are guaranteed not-null at this point.